src/plugins/vis_builder/public/application/components/data_tab/field_bucket.tsx

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from 'react';
+import {
+  EuiText,
+  EuiButtonIcon,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiSpacer,
+  EuiProgress,
+} from '@elastic/eui';
+import { i18n } from '@osd/i18n';
+
+import { IndexPatternField } from '../../../../../data/public';
+
+import { Bucket } from './types';
+import './field_bucket.scss';
+
+interface Props {
+  bucket: Bucket;
+  field: IndexPatternField;
+  onAddFilter: (field: IndexPatternField | string, value: string, type: '+' | '-') => void;
+}
+
+export function VisBuilderFieldBucket({ bucket, field, onAddFilter }: Props) {
+  const { count, display, percent, value } = bucket;
+  const { filterable: isFilterableField, name: fieldName } = field;
+
+  const emptyTxt = i18n.translate('visBuilder.fieldChooser.detailViews.emptyStringText', {
+    // We need this to communicate to users when a top value is actually an empty string
+    defaultMessage: 'Empty string',
+  });
+  const addLabel = i18n.translate(
+    'visBuilder.fieldChooser.detailViews.filterValueButtonAriaLabel',
+    {
+      defaultMessage: 'Filter for {fieldName}: "{value}"',
+      values: { fieldName, value },
+    }
+  );
+  const removeLabel = i18n.translate(
+    'visBuilder.fieldChooser.detailViews.filterOutValueButtonAriaLabel',
+    {
+      defaultMessage: 'Filter out {fieldName}: "{value}"',
+      values: { fieldName, value },
+    }
+  );
+
+  const displayValue = display || emptyTxt;
+
+  return (
+    <>
+      <EuiFlexGroup justifyContent="spaceBetween" responsive={false} gutterSize="s">
+        <EuiFlexItem className="vbFieldDetails__barContainer" grow={1}>
+          <EuiFlexGroup justifyContent="spaceBetween" gutterSize="xs" responsive={false}>
+            <EuiFlexItem grow={1} className="eui-textTruncate">
+              <EuiText
+                title={`${displayValue}: ${count} (${percent.toFixed(1)}%)`}
+                size="xs"
+                className="eui-textTruncate"
+              >
+                {displayValue}
+              </EuiText>
+            </EuiFlexItem>
+            <EuiFlexItem grow={false} className="eui-textTruncate">
+              <EuiText color="secondary" size="xs" className="eui-textTruncate">
+                {percent.toFixed(1)}%
+              </EuiText>
+            </EuiFlexItem>
+          </EuiFlexGroup>
+          <StringFieldProgressBar value={value} percent={percent} count={count} />
```

Comment:
It's more about nicely mapping the inputs to what `EUIProgress` supports. I suspect the initial reason is they intended to replace it with something better than a progress bar. I waffled back and forth a bit about whether it would be better to just inline it directly here.

Reply:
Moved it inline.